### DISCUSSION TITLE
Unable to run graphql-ws with graphql-compose

### DISCUSSION THREAD

**Discussion by ziedHamdi** [enisdenjo/graphql-ws#223] [Q&A] [answered]
**Expected Behaviour**
I expected it to have the arguments passed to resolvers

**Actual Behaviour**
but instead only the parent object is passed : resolve: async (parent, arg2, context) => {

**Debug Information**
I created a simple example to fill an issue in gqc https://github.com/graphql-compose/graphql-compose-subscription-boilerplate/issues/3

**Further Information**
I found the documentation on apollo confusing since their recommendation is to have query and mutation operations over http and only low bandwith consumption subscriptions over ws (creating a dynamic Link that detects the type of operation and uses ws only for subscriptions). On your docs you recommend using graphql-ws on all operations, which seams to me better since http is verbose and the overall bandwith consumption is bigger thrgough "traditional calls" if we already have an open socket. I think there should be more documentation about this choice

**Comment by enisdenjo** [comment, marked as answer]
Disclaimer: I have no experience with graphql-compose whatsoever.

To provide a contextValue to graphql resolvers, you have to supply the [`context` option](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.serveroptions.md#context). Please check the ["ws server usage with custom context value" recipe](https://github.com/enisdenjo/graphql-ws#context).

The `roots` option is a convenience that gets injected to the rootValue during graphql's execute or subscribe by operation type.
https://github.com/enisdenjo/graphql-ws/blob/f2a92aff2be9e700d1931f0c600c71b67146f617/src/server.ts#L738
In your case (from the referenced issue), the `roots` option can be completely omitted as it does nothing anyway.

`arg2` (aka `args`) are arguments provided to the field in the GraphQL query. You supply them from the client.

**Reply by ziedHamdi** [reply]
Ok thanks,

I'll check with the authors of  graphql-compose, I thought you might be interested in addressing your "customers" who use their solution as they have 408K weekly downloads
